corehq/apps/smsbillables/management/commands/bootstrap_twilio_gateway.py

```diff
@@ -7,101 +7,124 @@
-    args = ""
-    label = ""
+def bootstrap_twilio_gateway(orm):
+    currency_class = orm['accounting.Currency'] if orm else Currency
+    sms_gateway_fee_class = orm['smsbillables.SmsGatewayFee'] if orm else SmsGatewayFee
+    sms_gateway_fee_criteria_class = orm['smsbillables.SmsGatewayFeeCriteria'] if orm else SmsGatewayFeeCriteria
 
-    def handle(self, *args, **options):
-        # iso -> provider -> rate
-        def get_twilio_data():
-            twilio_file = open('corehq/apps/smsbillables/management/'
-                               'commands/pricing_data/twilio-international-sms-rates.csv')
-            twilio_csv = csv.reader(twilio_file)
-            twilio_data = {}
-            skip = 0
-            for row in twilio_csv:
-                if skip < 4:
-                    skip += 1
-                    continue
+    # iso -> provider -> rate
+    def get_twilio_data():
+        twilio_file = open('corehq/apps/smsbillables/management/'
+                           'commands/pricing_data/twilio-international-sms-rates.csv')
+        twilio_csv = csv.reader(twilio_file)
+        twilio_data = {}
+        skip = 0
+        for row in twilio_csv:
+            if skip < 4:
+                skip += 1
+                continue
+            else:
+                try:
+                    iso = row[0].lower()
+                    provider = row[1].split('-')[1].lower().replace(' ', '')
+                    rate = float(row[2])
+                    if not(iso in twilio_data):
+                        twilio_data[iso] = {}
+                    twilio_data[iso][provider] = rate
+                except IndexError:
+                    logger.info("Twilio index error %s:" % row)
+        twilio_file.close()
+        return twilio_data
+
+    # iso -> provider -> (country code, number of subscribers)
+    def get_mach_data():
+        mach_workbook = xlrd.open_workbook('corehq/apps/smsbillables/management/'
+                                           'commands/pricing_data/Syniverse_coverage_list_DIAMONDplus.xls')
+        mach_table = mach_workbook.sheet_by_index(0)
+        mach_data = {}
+        try:
+            row = 7
+            while True:
+                country_code = int(mach_table.cell_value(row, 0))
+                iso = mach_table.cell_value(row, 1)
+                network = mach_table.cell_value(row, 5).lower().replace(' ', '')
+                subscribers = 0
+                try:
+                    subscribers = int(mach_table.cell_value(row, 10).replace('.', ''))
+                except ValueError:
+                    logger.info("Incomplete subscriber data for country code %d" % country_code)
+                if not(iso in mach_data):
+                    mach_data[iso] = {}
+                mach_data[iso][network] = (country_code, subscribers)
+                row += 1
+        except IndexError:
+            pass
+        return mach_data
+
+    twilio_data = get_twilio_data()
+    mach_data = get_mach_data()
```

Comment:
ok - I just want to make sure we're not mixing rate data - can we make sure we're not pulling mach rate data into the twilio rates?